7.0 nl ing mt940

NL66278 opened the pull request:
Update mt940 import for Dutch bank ING to latest specs.

NL66278 committed:
[ENH] Update mt940 import for dutch ing bank.

NL66278 committed:
[ENH] Update mt940 import for dutch ing bank.

NL66278 committed:
[FIX] Some changes for flake8 and pylint.

NL66278 committed:
Merge branch '7.0_nl_ing_mt940' of github.com:nl66278/banking into 7.0_nl_ing_mt940

Conflicts:
	account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py

sbidoul commented on account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py:
```diff
@@ -58,7 +63,9 @@ def handle_tag_60F(self, cr, data):
         super(IngMT940Parser, self).handle_tag_61(cr, data)
-        parsed_data = self.tag_61_regex.match(data).groupdict()
+        re_61 = self.tag_61_regex.match(data)
+        assert re_61, 'Cannot parse %s' % data
```
Since this can reveal a input data format error, I'd suggest a raise instead of assert

sbidoul commented:
There are a few flake8 and pylint issues.

NL66278 committed:
[FIX] Change account_banking_nl_ing_mt940 to
    comply with flake and pylint.

NL66278 committed:
[FIX] Use raise ValueError instead of assert for error in import data.

NL66278 committed:
[FIX] Some more flake8 corrections in account_banking_nl_ing_mt940.
    It seems travis flake8 is more critical than pyflakes.

NL66278 commented:
@sbidoul Hello Stephane, I replaced the assert with a raise ValueError and fixed the flake/pylint issues.

hbrunn committed:
[FIX] don't pass column name as sql parameters

Commit:
[FIX] Account number column too short for some IBANs

Commit:
Add FR translation on account_payment_partner/sale/purchase

NL66278 committed:
Merge pull request #94 from hbrunn/7.0-fix-migratio into 7.0_nl_ing_mt940

NL66278 committed:
Merge pull request #104 from akretion/7.0-account-payment-fr-translation
    into 7.0_nl_ing_mt940

NL66278 committed:
Merge pull request #100 from
    StefanRijnhart/7.0-fix_account_number_column_length into 7.0_nl_ing_mt940

hbrunn commented on account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py:
```diff
@@ -102,8 +113,8 @@ def handle_tag_86(self, cr, data):
-            self.current_transaction.message = '/'.join(
-                filter(lambda x: bool(x), subfields['REMI']))
+            self.current_transaction.message = (
+                '/'.join([x for x in subfields['REMI'] if x]))
```
The beauty of the expression you replaced is that it doesn't create a temporary copy of the list. You can have the same effect if you change the list comprehension to a generator expression.

TLDR: remove outermost square brackets

hbrunn commented on account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py:
```diff
@@ -57,14 +63,19 @@ def handle_tag_60F(self, cr, data):
             self.current_statement.id)
 
     def handle_tag_61(self, cr, data):
+        """Parse 61F tag containing transaction data."""
```
it's tag 61 (without 'F'), right?

hbrunn commented on account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py:
```diff
@@ -57,14 +63,19 @@ def handle_tag_60F(self, cr, data):
         self.current_transaction.reference = parsed_data['reference']
 
     def handle_tag_86(self, cr, data):
+        """Parse 86F tag containing reference data."""
```
also no 'F' here?

hbrunn commented on account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py:
```diff
@@ -20,36 +21,41 @@
     name = _('ING MT940 (structured)')
     country_code = 'NL'
     code = 'INT_MT940_STRUC'
+    footer_regex = '^-}$'
```
I used https://www.ing.nl/media/ING_ibp_mt940_942_format_description_the_netherlands_tcm162-45288.pdf as reference, there they state the message trailer is '-XXX' (p6) - did they release another document replacing this one?

hbrunn commented on account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py:
```diff
@@ -20,36 +21,41 @@
     tag_61_regex = re.compile(
-        r'^(?P<date>\d{6})(?P<sign>[CD])(?P<amount>\d+,\d{2})N(?P<type>.{3})'
-        r'(?P<reference>\w{1,16})')
+        r'^(?P<date>\d{6})(?P<line_date>\d{4})'
```
Also from the document above, I don't see a line date there. You probably used some real test file that contains this extra field, would it be an option to make it optional in the regex (`\d{0,4}`)? I doesn't seem to be used anyways

NL66278 commented:
Holger:
I remove the F's where they should not be, I will look into the generator / lambda thing (your first remark).
The format indeed changed since you downloaded the document mentioned. Maybe I could put in some test/options to support the old format as well.

NL66278 committed:
[ENH] Support both old and new format for import.

In the last part of 2014 the ING bank changed the import format
for mt940 structured files. The new format was already supported. With
a few trivial changes we are now also able to import older files.

Also added some improvements/corrections suggested by Holger Brunn.

NL66278 commented:
Superseded by: https://github.com/OCA/bank-payment/pull/142
